packaging/builder-images/Dockerfile.x86_64-unknown-linux-musl

```diff
@@ -0,0 +1,29 @@
+ARG RUST_VERSION
+
+# update the version here as needed
+# via https://hub.docker.com/_/rust
+FROM rust:${RUST_VERSION}-buster
+
+# install musl
+# this setup borrowed from https://github.com/rust-embedded/cross/blob/v0.2.0/docker/musl.sh#L44
+# not using cross's default image there directly since that has older gcc (6.4)
+# which does not work for our dependencies like snmalloc
+RUN temp_dir=$(mktemp -d) \
+  && cd $temp_dir \
+  && curl -L https://github.com/richfelker/musl-cross-make/archive/v0.9.9.tar.gz | tar --strip-components=1 -xz \
+  && make install -j$(nproc) \
+    GCC_VER=9.2.0 \
+    MUSL_VER=1.2.0 \
+    DL_CMD="curl -C - -L -o" \
+    OUTPUT=/usr/local/ \
+    TARGET=x86_64-linux-musl \
+  && rm -rf $temp_dir
+
+ENV CARGO_TARGET_X86_64_UNKNOWN_LINUX_MUSL_LINKER=x86_64-linux-musl-gcc \
+    CC_x86_64_unknown_linux_musl=x86_64-linux-musl-gcc \
+    CXX_x86_64_unknown_linux_musl=x86_64-linux-musl-g++
+
+RUN rustup target add x86_64-unknown-linux-musl
```

Comment:
filed an issue on cross upstream to see if we can bump up the gcc version on the default cross images:
https://github.com/rust-embedded/cross/issues/432

If we can do that, we can derive this image from ` rustembedded/cross:x86_64-unknown-linux-musl` and not have to install musl ourselves.